platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformComponentBuilder.java

```diff
@@ -892,4 +895,35 @@ public Gossip buildGossip() {
         }
         return gossip;
     }
+
+    /**
+     * Provide a state hasher in place of the platform's default state hasher.
+     *
+     * @param stateHasher the state hasher to use
+     * @return this builder
+     */
+    @NonNull
+    public PlatformComponentBuilder withStateHasher(@NonNull final StateHasher stateHasher) {
+        throwIfAlreadyUsed();
+        if (this.stateHasher != null) {
+            throw new IllegalStateException("Signed state hasher has already been set");
+        }
+        this.stateHasher = Objects.requireNonNull(stateHasher);
+        return this;
+    }
+
+    /**
+     * Build the state hasher if it has not yet been built. If one has been provided via
+     * {@link #withStateHasher(StateHasher)}, that hasher will be used. If this method is called more than once, only
+     * the first call will build the state hasher. Otherwise, the default hasher will be created and returned.
+     *
+     * @return the signed state hasher
+     */
+    @NonNull
+    public StateHasher buildStateHasher() {
```

Comment:
For the `PlatformComponentBuilder` is a builder for `Platform`:

```
Platform platform = new PlatformComponentBuilder(foo).withStatusStateMachine(bar).build();
```

I do not see why this method is `public`?

Reply:
Should not be public, but currently our package layout doesn't allow non-public methods. I agree that the way you suggest is the correct way, but first we need to reorganize our package structure.